badge_server/main.py

```diff
@@ -76,518 +58,196 @@ def _get_pair_status_for_packages(pkg_sets):
-                max_level = level
-            dep_detail['installed_version'] = dep.installed_version
-            dep_detail['latest_version'] = dep.latest_version
-            dep_detail['priority'] = dep.priority.level.name
-            dep_detail['detail'] = dep.priority.details
-            details[dep.name] = dep_detail
-            res['status'] = max_level.name
-            res['details'] = details
-            res['deprecated_deps'] = deprecated_deps
-        res['timestamp'] = datetime.datetime.now().strftime(
-            badge_utils.TIMESTAMP_FORMAT)
-
-        # Write the result to Cloud Datastore
-        cache.set('{}_dependency_badge'.format(package_key), res)
-
-    if not utils._is_package_in_whitelist([package_name]):
-        dependency_res = badge_utils._build_default_result(
-            badge_type=badge_utils.BadgeType.DEP_BADGE,
-            status='UNKNOWN',
-            details={})
-    else:
-        dependency_res = cache.get('{}_dependency_badge'.format(package_key))
-
-    if dependency_res is None:
-        details = badge_utils.DEFAULT_DEPENDENCY_RESULT
-    else:
-        details = dependency_res
-
-    # Run the check if dependency_res is None or forced to populate the cache
-    # or the cache is outdated.
-    if dependency_res is None or force_run_check is not None:
-        threading.Thread(target=run_check).start()
-    elif dependency_res is not None:
-        timestamp = dependency_res.get('timestamp')
-        if not badge_utils._is_github_cache_valid(timestamp):
-            threading.Thread(target=run_check).start()
-
-    badge = badge_utils._get_badge(details, badge_name)
-    response = flask.make_response(badge)
-    response.content_type = badge_utils.SVG_CONTENT_TYPE
-    response.headers['Cache-Control'] = 'no-cache'
-    response.add_etag()
+    # TODO: implement checks for the following conditions
+    # (they are currently using dummy info)
+    is_pypi = compat_utils._is_package_in_whitelist([pkgname])
+    is_latest_commit = True
+    if 'github.com' in pkgname:
+        parens_text = 'master' if is_latest_commit else 'updating...'
+        bdgname = '{} ({})'.format(bdgname, parens_text)
+    elif is_pypi:
+        bdgname = '{} (PyPI)'.format(bdgname)
 
-    return response
+    return bdgname
 
 
-@app.route('/self_dependency_badge_target')
-def self_dependency_badge_target():
-    """Return a dict that contains dependency status and details."""
-    package_name = flask.request.args.get('package')
-    result_dict = _get_result_from_cache(
-        package_name=package_name,
-        badge_type=badge_utils.BadgeType.DEP_BADGE)
+@app.route('/one_badge_image')
+def one_badge_image():
+    """Generate a badge that captures all checks."""
+    pkgname = flask.request.args.get('package')
+    bdgname = flask.request.args.get('badge')
+
+    commit_number = badge_utils._calculate_commit_number(pkgname)
+    status, timestamp, _, _, _ = _get_results_from_compatibility_store(pkgname)
```

Comment:
I don't think that you use `timestamp` - are we not using a linter that would catch that?

Reply:
removed timestamp and updated functions